### PR TITLE
Sidebar: New Layers Panel

### LINKED ISSUE #10804: Sidebar: Move Layers panel out of the Sidebar
## Feature Description

PR should be based on/merged to the `sidebar` branch.

### PR DESCRIPTION
**Pull request by merapi**
## Context

Part of editor redesign. 

## Summary

Lifts the layers panel from the inspector pane (which was combined with the design pane to create the new sidebar) and puts it opposite the sidebar as part of the editor's footer. 

## Relevant Technical Choices

Added a few more props to the shared `ToggleButton` that's used for the many popups of the footer to add a variant with copy visible/no icon and prevent tooltip from showing. 

Added `<DirectionAware>` to the footer so that the padding there is consistent and the layers panel wouldn't require additional work to be readable in RTL. I think it looks better now. 

## To-do

- [x] Fix tests
- [x] Adjust colors
- [x] Check a11y

Opted against moving all of the layers panel files into their own directory (following the suit of the helpcenter, keyboard nav and checklist) because it makes it harder to grok the updates.

Talked to @agingoldseco about the new toggle design, we're going to do that [later as a nonblocking issue.](https://github.com/GoogleForCreators/web-stories-wp/issues/10891)

The layer panel accessibility is not an issue here, it would be introducing new use patterns should we expose it to the keyboard at this point which would bloat the scope significantly. It was discussed at length 5 months ago in the design slack channel. There are keyboard shortcuts for everything that the layers panel can do. If we decide that now the layers panel should be keyboard accessible that would be its own scope of work and own item on the roadmap. 

## User-facing changes

| LTR | RTL | 
| --- | --- | 
| Screen Shot 2022-03-11 at 11 23 36 AM | Screen Shot 2022-03-11 at 11 24 13 AM |

With Meta boxes 
| LTR | RTL | 
| --- | --- | 
| Screen Shot 2022-03-11 at 1 38 26 PM | Screen Shot 2022-03-11 at 1 38 37 PM |

## Testing Instructions

Use the layers panel, now on the opposite side as the sidebar. 
You should be able to expand and collapse it with cursor and keyboard.
Confirm layer selection and behavior.
Check LTR and RTL. 
Install and activate 'yoast seo' to see meta box show up. 

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10804

### CODE REVIEW

File: packages/story-editor/src/components/secondaryPopup/index.js
Comment (merapi) [inline packages/story-editor/src/components/secondaryPopup/index.js]: I had to set a higher z-index here (didn't want to change z-indexes in Page Side Menu) in order to prevent this:

https://user-images.githubusercontent.com/21036224/157312306-b299a49e-8688-4822-9b83-754525af740b.mov

File: packages/story-editor/src/components/footer/primaryMenu.js
Comment (barklund) [inline packages/story-editor/src/components/footer/primaryMenu.js:56]: If you have metaboxes (from other plugins), the alignment is a bit off (between layers button and layers panel).

IMO the layers panel button should be last (right-most) and the metaboxes button should be moved a bit, but UX might have a better suggestion.

Screen Shot 2022-03-08 at 15 37 31

Comment (merapi) [reply packages/story-editor/src/components/footer/primaryMenu.js:56]: I've noticed that panels on the left side of the footer are not always aligned to the button:

Checklist:
image

Shortcuts:
image

Not sure why - maybe to save the canvas overflow? that might be especially useful for the Layers panel that will be visible more often than the Shortcuts (so we want to cover as little of the canvas as possible).
I will ask Design pod about that.

File: packages/story-editor/src/components/secondaryPopup/index.js
Comment (miina) [inline packages/story-editor/src/components/secondaryPopup/index.js]: In the design-system `Popup`, anchor is a `ref` of the element that has the popup and `spacing` in the format of `{ x, y }` is used for spacing and `placement` is used to define if it's on the left (or somewhere else) relative to the anchor. Could we use the same logic here for consistency? E.g. could we use `spacing` with the relevant values and/or `placement` instead of `anchor`. Thoughts?

File: packages/story-editor/src/components/panels/design/layer/layerPanel.js
Comment (miina) [inline packages/story-editor/src/components/panels/design/layer/layerPanel.js]: Is the `HEAD_AREA_HEIGHT` related to the WordPress admin menu? Or does it include the WP admin menu + the buttons? Or just buttons?

In case it includes the WP admin menu, the value should come from `useConfig`,  e.g.
```
const { styleConstants: { topOffset } = {} } = useConfig();
```

If it's the menu area with buttons, let's use the `HEADER_HEIGHT` from the general constants file instead.

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (miina) [inline packages/story-editor/src/components/footer/layers/layers.js]: Hm, we should really use a button from the design-system here -- whenever there's a button without variant in Figma, would be good to confirm with UX to check which button it is. The only thing that I can see in Figma right now is that it's a rectangle button of size small, unfortunately variant hasn't been defined.

Otherwise we'd need to also add hover/focus etc. here separately.
From [Figma](https://www.figma.com/file/Ha6kk9yXDN1rB80NzcDf2J/_Editor---Design-System-%22Style-Guide%22?node-id=3821%3A81609), it looks like it's a quaternary button, let's use that instead. Then we also wouldn't probably need to style the `LabelText` separately.

Comment (merapi) [reply packages/story-editor/src/components/footer/layers/layers.js]: It makes sense, I've looked at other buttons in the footer and they all didn't match this one so I've started to create my own style (just got the permissions to access this Figma file you linked) - I will see what can be done here :)

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [inline packages/story-editor/src/components/footer/layers/layers.js]: setting the zIndex here to 9999 is alot. I'm addressing the zIndex here: https://github.com/GoogleForCreators/web-stories-wp/pull/10861 - following this format would be very helpful. i'm currently blocked by Karma tests from merging it. You can follow the same pattern as the other footer popups. I'll write up more specifically what this is here shortly.

File: packages/story-editor/src/components/footer/primaryMenu.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/primaryMenu.js:56]: is this in figma?

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/layers/layers.js]: So this is actually more in line with the other popups in the footer: 
Screen Shot 2022-03-10 at 10 53 16 AM

Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/layers/layers.js]: The only reason the caret is on the zoom is because it's a dropdown(up) menu. I was able to reuse the `ToggleButton` that the rest of the footer popups utilize with only a few modifications. I'll screenshot the diff here. 

I'd like to request you just leave the z index be and not further tinker with it. It's been really hard to untangle and my PR will resolve your issues with it. 
Screen Shot 2022-03-10 at 10 55 36 AM
Screen Shot 2022-03-10 at 10 55 25 AM

Comment (miina) [reply packages/story-editor/src/components/footer/layers/layers.js]: Good point that it's actually a Toggle!

Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/layers/layers.js]: Amy might be fine with the design change to cater to consistency. 

That said, if one notification bubble has a bg and a circle then they _all_ should and that could be its own lower priority ticket.

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (barklund) [reply packages/story-editor/src/components/footer/layers/layers.js]: `Number.MAX_SAFE_INTEGER` **FTW** 🚀 🌔 .

File: packages/story-editor/src/components/footer/primaryMenu.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/primaryMenu.js:56]: I will talk to Amy about alignment.

File: packages/story-editor/src/components/footer/footerLayout.js
Comment (BrittanyIRL) [inline packages/story-editor/src/components/footer/footerLayout.js:56]: Added `<DirectionAware>` to make spacing consistent in RTL, otherwise the layers panel toggle is right up against the edge of the browser

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [inline packages/story-editor/src/components/footer/layers/layers.js]: `SecondaryPopup` was created with the left (in LTR) side of the footer in mind, so everything is aligned to the left. This is quick and dirty way to align to the right side. It's the only one that does this so I'd rather not dirty up the shared popup for 1 case

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [inline packages/story-editor/src/components/footer/layers/layers.js:34]: ensures scroll

File: packages/story-editor/src/components/panels/design/layer/layerPanel.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/panels/design/layer/layerPanel.js]: was able to ditch this because the shared popup is accounting for available height already.

File: packages/story-editor/src/components/secondaryPopup/index.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/secondaryPopup/index.js]: We should be good here now. @merapi 's idea to add anchor to control the left vs right value was great, that combined with an override in the layer set up and we're gucci. No need to further tweak anything.

File: packages/story-editor/src/components/footer/primaryMenu.js
Comment (merapi) [reply packages/story-editor/src/components/footer/primaryMenu.js:56]: Here is the Slack discussion: https://xwp.slack.com/archives/C01ESBB8ZN1/p1646935295628609?thread_ts=1646930940.483089&cid=C01ESBB8ZN1

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (merapi) [reply packages/story-editor/src/components/footer/layers/layers.js]: Just pasting the last input from Amy:

image

(so Q-styled button with up/down arrow).
But now since you made the change maybe just double-check with her.

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (merapi) [reply packages/story-editor/src/components/footer/layers/layers.js]: I thought `9999` is some kind of magic number, we use it a lot 😉  (I guess it just spread like a virus)

image

File: packages/story-editor/src/components/footer/primaryMenu.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/primaryMenu.js:56]: Talked to amy about what I have for this and she's fine with it as is. Alignment is either to the left or the right, not the center ever. Looks good with and without meta buttons - I added screenshots to the description.

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [reply packages/story-editor/src/components/footer/layers/layers.js]: Chatted with amy, thanks for the screenshot of your chat it's helpful (slack won't open the links). 

we're going to handle the chevron/updates from the existing toggle button separate to unblock work and be consistent with other buttons with same behavior. (#10891 )

File: packages/story-editor/src/components/secondaryPopup/index.js
Comment (miina) [reply packages/story-editor/src/components/secondaryPopup/index.js]: Hm, still not convinced about the naming.

Since `anchor` is a `ref` in the "default" `Popup` then could we for consistency use different naming for the placement here? E.g. why not use `placement` instead as in the default `Popup`? 🤔 Or `anchorPlacement` at least?

The logic itself seems all good but just the naming is inconsistent with the general `Popup`. Thoughts?

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (maxyinger) [reply packages/story-editor/src/components/footer/layers/layers.js]: we could also do a default prop in `NavigationWrapper` that would allow us to not have to edit any other instance like so:
```
const NavigationWrapper = styled.div`
  ...
  ${({ alignRight = false }) => (alignRight ? 'right: 0' : 'left: 0')}
```
But I'm cool with this as is since it's pretty isolated and straight forward.

A third option here that I think would be the equivalent is just doing:
```
const StyledNavigationWrapper = styled(NavigationWrapper)`
  right: 0;
  left: auto;
```

File: packages/story-editor/src/components/panels/design/layer/layer.js
Comment (maxyinger) [inline packages/story-editor/src/components/panels/design/layer/layer.js:142]: apologies for the styling here being a little complicated 😅 

Looks like something is out of sync here tho with the hover styles over the layer actions:
| Staging | This Branch |
| --- | ---|
| Screen Shot 2022-03-14 at 9 04 51 AM |  Screen Shot 2022-03-14 at 9 05 03 AM |

Looks like it is do to the background color of the popup changing.

**Review by maxyinger — APPROVED** [review]

File: packages/story-editor/src/components/secondaryPopup/index.js
Comment (samwhale) [reply packages/story-editor/src/components/secondaryPopup/index.js]: I'll update this to `placement`

File: packages/story-editor/src/components/footer/layers/layers.js
Comment (BrittanyIRL) [inline packages/story-editor/src/components/footer/layers/layers.js:50]: like the `alignRight` update

**Review by BrittanyIRL — APPROVED** [review]
Thanks for getting this finished, @samwhale !!

**Review by samwhale — APPROVED** [review]

**Pull request merged**
